packages/gatsby/src/datastore/in-memory/indexing.ts

```diff
@@ -21,6 +24,88 @@ export type FilterOp =  // TODO: merge with DbComparator ?
 // Note: `undefined` is an encoding for a property that does not exist
 
 export type FilterCacheKey = string
+type GatsbyNodeID = string
+
+export interface IGatsbyNodePartial {
+  isGatsbyNodePartial: boolean
+  id: GatsbyNodeID
+  internal: {
+    counter: number
+  }
+  indexFields: Set<string>
+  [k: string]: any
+}
```

Comment:
We could if others feel strongly about it. Having `indexFields` as a default key on the public node object feels like we'd be exposing unnecessary implementation details. If we ever need to add more to the internals, I'd not want to have to expose those as well.

Reply:
We are creating partial object from `IGatsbyNode`, so if that changes (in a way that impact this, we will know because types will not be happy and query running tests start breaking). They also are different in more ways - fields are not nested like on original node, rather they are flattened fields with `nested.field` style keys in root of partial.

They really are not the same and trying to share types in this case doesn't make sense to me, as we have to employ ignoring things